bot/seasons/evergreen/snakes/utils.py

```diff
@@ -435,8 +436,9 @@ def startup_event_check(reaction_: Reaction, user_: Member):
                 if reaction.emoji == JOIN_EMOJI:
                     await self.player_join(user)
                 elif reaction.emoji == CANCEL_EMOJI:
-                    if self.ctx.author == user:
-                        await self.cancel_game(user)
+                    if user == self.author or all((self._is_moderator(user), user not in self.players)):
```

Comment:
Using `all()` feels more confusing for this when can just be a simple `and` between two objects.
```suggestion
                    if user == self.author or (self._is_moderator(user) and user not in self.players):
```